### PR TITLE
Improvement/artesca 10990 expose account role selector with module federation

### PR DESCRIPTION
**Pull request by MonPote**

### CODE REVIEW

File: src/react/ui-elements/SelectAccountIAMRole.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+import { LatestUsedCapacity } from '../next-architecture/domain/entities/metrics';
+import {
+  AccessibleAccountsAdapterProvider,
+  useAccessibleAccountsAdapter,
+} from '../next-architecture/ui/AccessibleAccountsAdapterProvider';
+import { AccountsLocationsEndpointsAdapterProvider } from '../next-architecture/ui/AccountsLocationsEndpointsAdapterProvider';
+import { getListRolesQuery } from '../queries';
+import { regexArn } from '../utils/hooks';
+
+class NoOppMetricsAdapter implements IMetricsAdapter {
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/SelectAccountIAMRole.tsx]:
```suggestion
class NoOpMetricsAdapter implements IMetricsAdapter {
```

File: src/react/ui-elements/SelectAccountIAMRole.tsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      !hideAccountRoles.find(
+        (hideRole) =>
+          hideRole.accountName === accountName &&
+          hideRole.roleName === role.RoleName,
+      ),
+  );
+};
+
+/**
+ * DataServiceRoleProvider is using the the path to figure out what is the current account.
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/SelectAccountIAMRole.tsx]:
```suggestion
 * DataServiceRoleProvider is using the path to figure out what is the current account.
```

File: src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    await waitFor(() => {
+      expect(seletors.roleSelect()).toBeInTheDocument();
+    });
+
+    await userEvent.click(seletors.roleSelect());
+    await userEvent.type(seletors.roleSelect(), 'data-consumer');
+
+    expect(screen.getByText(/no options/i)).toBeInTheDocument();
+
+    debug();
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx]:
```suggestion
    
```

File: src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+  beforeAll(() => {
+    server.listen({ onUnhandledRequest: 'error' });
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  afterAll(() => {
+    server.close();
+  });
+  it('renders with normal props', async () => {
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx:254]: I think you want an additional test where you change account and change roles for this account and check if the options of the role select matches

File: src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        />
+      </LocalWrapper>,
+    );
+    await waitFor(() => {
+      expect(seletors.accountSelect()).toBeInTheDocument();
+    });
+
+    expect(screen.getByText('no-bucket')).toBeInTheDocument();
+
+    debug();
```
Comment (ChengYanJin) [inline src/react/ui-elements/__tests__/SelectAccountIAMRole.test.tsx]: Can remove this debug( ) ;)

File: src/react/ui-elements/SelectAccountIAMRole.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+ * We assume the user won't have an account with this name.
+ */
+const INTERNAL_DEFAULT_ACCOUNT_NAME_FOR_INITIALIZATION =
+  '__INTERNAL_DEFAULT_ACCOUNT_NAME_FOR_INITIALIZATION__';
+
+const AssumeDefaultIAMRole = ({
+  defaultValue,
+}: Pick<SelectAccountIAMRoleWithAccountProps, 'defaultValue'>) => {
+  const accessibleAccountsAdapter = useAccessibleAccountsAdapter();
+  const metricsAdapter = new NoOppMetricsAdapter();
```
Comment (ChengYanJin) [inline src/react/ui-elements/SelectAccountIAMRole.tsx]: seems you forgot to import this ?

**Review by ChengYanJin — APPROVED** [review]
LGTM! Yohoo!!!

**Pull request merged**